pkg/cleanup/cleanup.go

```diff
@@ -0,0 +1,52 @@
+package cleanup
+
+import (
+	"github.com/containous/maesh/pkg/k8s"
+	"github.com/sirupsen/logrus"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+// Cleaner is an interface for the cleanup methods.
+type Cleaner interface {
+	CleanShadowServices() error
+}
+
+// Ensure the Prepare fits the Preparer interface
+var _ Cleaner = (*Cleanup)(nil)
+
+// Cleanup holds the clients for the various resource controllers.
+type Cleanup struct {
+	client k8s.Client
+	log    logrus.FieldLogger
+}
+
+// NewCleanup returns an initialized cleanup object.
+func NewCleanup(log logrus.FieldLogger, client k8s.Client) Cleaner {
+	return &Cleanup{
+		client: client,
+		log:    log,
+	}
+}
+
+// CleanShadowServices deletes all shadow services from the cluster.
+func (c *Cleanup) CleanShadowServices() error {
+	serviceList, err := c.client.GetKubernetesClient().CoreV1().Services(metav1.NamespaceAll).List(metav1.ListOptions{
+		LabelSelector: "app=maesh",
+	})
+	if len(serviceList.Items) == 0 {
+		// No services found, return without error.
+		return nil
+	}
+
+	if err != nil {
```

Comment:
Yep, we should check for errors before checking for an empty response, otherwise in the case of a wrong kubeconfig or wrong kubernetes API address for example, users might think that there is nothing to be cleaned and it was successful, while they just provided invalid parameters.